config/config_remote_write.go

```diff
@@ -0,0 +1,162 @@
+package config
+
+import (
+	"errors"
+	"fmt"
+	"sort"
+	"strings"
+)
+
+// TODO(bwplotka): Consider an util for "preference enums" as it's similar code for rw compression, proto type and scrape protocol.
+
+// RemoteWriteProtoType represents the supported protobuf types for the remote write.
+type RemoteWriteProtoType string
```

Comment:
ProtoType feels awkward, can we call it `RemoteWriteProtoVersion` ?

Reply:
I tried ProtoMessages, ProtoVersion etc but they might be worse options.

ProtoVersion feels like Protocol Version, so spec version? But it represent protobuf type, to in short prototype, not protocol version and I want to make it explicit. Or do you think ProtoVersion would not mislead?

Perhaps `RemoteWriteProtobufType` would be better?

Reply:
`RemoteWriteProtobufType` or `RemoteWriteProtobufVersion`, or maybe `RemoteWriteWireType/Format/Version` since we've discussed implementing and allow usage of something like Arrow in the future?